src/v/archival/archival_policy.cc

```diff
@@ -45,6 +45,32 @@ bool archival_policy::eligible_for_compacted_reupload(
     return s.finished_self_compaction();
 }
 
+std::ostream&
+operator<<(std::ostream& os, upload_candidate::creation_status cs) {
+    switch (cs) {
+    case upload_candidate::creation_status::success:
+        return os << "success";
+    case upload_candidate::creation_status::err_size_unchanged:
+        return os << "upload candidate size unchanged for compacted reupload";
+    case upload_candidate::creation_status::err_offset_inside_batch:
+        return os << "candidate begin/end offset inside batch boundary";
+    case upload_candidate::creation_status::err_no_segments:
```

Comment:
err_no_segments and err_no_segment are too similar in name but have very different messages. We should encode more meaning in the name. There is some implicit encoding that some of these only apply to compacted segments, others to non-compacted but when using the values without looking at the `<<` implementation it is impossible to distinguish which is which.